x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/PhysicalOperationProviders.java

```diff
@@ -21,6 +23,7 @@ interface PhysicalOperationProviders {
     PhysicalOperation groupingPhysicalOperation(
         AggregateExec aggregateExec,
         PhysicalOperation source,
-        LocalExecutionPlannerContext context
+        LocalExecutionPlannerContext context,
+        List<EsPhysicalOperationProviders.ShardContext> shardContexts
```

Comment:
It's feels a bit odd to provide an implementation piece of EsPhysicalOperationProviders (in the list) to the interface, given that actually `EsPhysicalOperationProviders`, which implements this if'ace, already has the list of `shardContexts`. I wonder if we could embbed it into the `LocalExecutionPlannerContext` itself, since it's local execution specific anyways.
Maybe it should be passed it to `LocalExecutionPlaner#plan` from `ComputeService#runCompute`, where the list of shard contexts is already available?

Reply:
That makes sense - I did that change in 2a8e987c579e662b18a093db7d1319957bd52ab7.